jetty-core/jetty-compression/jetty-compression-api/src/main/java/org/eclipse/jetty/compression/Compression.java

```diff
@@ -0,0 +1,355 @@
+     *
+     * @return the HttpField for {@code Content-Encoding}.
+     */
+    public abstract HttpField getContentEncodingField();
+
+    public abstract DecoderConfig getDefaultDecoderConfig();
+
+    public abstract void setDefaultDecoderConfig(DecoderConfig decoderConfig);
+
+    public abstract EncoderConfig getDefaultEncoderConfig();
+
+    public abstract void setDefaultEncoderConfig(EncoderConfig encoderConfig);
+
+    /**
+     * The name of the encoding if seen in the HTTP protocol in fields like {@code Content-Encoding}
+     * or {@code Accept-Encoding}.  This name is also reused for the {@code ETag} representations
+     * of the compressed content.
+     *
+     * @return the name of the Content-Encoding for this compression implementation.
+     */
+    public String getEncodingName()
+    {
+        return encodingName;
+    }
+
+    /**
+     * Get the ETag suffix.
+     *
+     * @return the etag suffix for this compression.
+     */
+    public String getEtagSuffix()
+    {
+        return etagSuffix;
+    }
+
+    /**
+     * The filename extensions for this compression implementation.
+     *
+     * <p>
+     * Not an exhaustive list, just the most commonly seen extensions.
+     * </p>
+     *
+     * @return the list of common extension names (all lowercase) for this compression implementation.
+     * ordered by most common to least common.
+     */
+    public abstract List<String> getFileExtensionNames();
+
+    public abstract int getMinCompressSize();
+
+    public abstract void setMinCompressSize(int minCompressSize);
+
+    /**
+     * @return the name of the compression implementation.
+     */
+    public abstract String getName();
+
+    /**
+     * The {@link HttpField} for {@code X-Content-Encoding} suitable for this Compression implementation.
+     *
+     * @return the HttpField for {@code X-Content-Encoding}.
+     */
+    public abstract HttpField getXContentEncodingField();
+
+    /**
+     * Create a new {@link InputStream} to decode (decompress) with this compression implementation.
+     *
+     * @param in the input stream to write the decoded (decompressed) bytes to
+     * @return the {@link InputStream} implementation for this compression.
+     * @throws IOException if unable to create InputStream
+     */
+    public InputStream newDecoderInputStream(InputStream in) throws IOException
+    {
+        return newDecoderInputStream(in, getDefaultDecoderConfig());
+    }
+
+    /**
+     * Create a new {@link InputStream} to decode (decompress) with this compression implementation.
+     *
+     * @param in the input stream to write the decoded (decompressed) bytes to
+     * @param config the {@link DecoderConfig} for this input stream.
+     * @return the {@link InputStream} implementation for this compression.
+     * @throws IOException if unable to create InputStream
+     */
+    public abstract InputStream newDecoderInputStream(InputStream in, DecoderConfig config) throws IOException;
```

Comment:
This uses the compression implementation version of the InputStream, not the wrapped version of a wrapped of an extension.  It's mostly used for tests and demonstration purposes.  It will come to be especially handy for Jetty Client users in the future (when this is integrated there) as well.